llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9698,6 +9698,11 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
     // beginning of the dedicated latch block.
     auto *OrigExitingVPV = PhiR->getBackedgeValue();
     auto *NewExitingVPV = PhiR->getBackedgeValue();
+    // Don't add selects here for partial reductions because the phi and partial
+    // reduction values have less vector elements than Cond. But, each operand
+    // in a select instruction needs to have the same number of vector elements,
+    // so the compiler would crash. Instead, a select, with the active lane
+    // mask, is applied to the inputs to the partial reduction.
```

Comment:
This could probably be more concise, as something like "The output of a partial reductions has fewer lanes than the VF so don't emit a select, as its operands would have differing numbers of lanes. A partial reduction masks its input instead."